servo-build-dependencies/android.sls

```diff
@@ -79,51 +80,24 @@ android-sdk-current:
 
 
 android-ndk:
-  file.managed:
-    - name: {{ common.servo_home }}/android/ndk/{{ android.ndk.version }}/android-ndk-{{ android.ndk.version }}-linux-x86_64.bin
-    - source: https://dl.google.com/android/ndk/android-ndk-{{ android.ndk.version }}-linux-x86_64.bin
+  archive.extracted:
+    - name: {{ common.servo_home }}/android/ndk/{{ android.ndk.version }}/android-ndk-{{ android.ndk.version }}-linux-x86_64.zip
+    - source: https://dl.google.com/android/repository/android-ndk-{{ android.ndk.version }}-linux-x86_64.zip
     - source_hash: sha512={{ android.ndk.sha512 }}
+    - archive_format: zip
     - user: servo
```

Comment:
Also pass `archive_user` (see the `android-sdk` state for why, feel free to copy the justification comment down here as well.)